src/metrics/filters/tags_filter.go

```diff
@@ -39,11 +39,18 @@ var (
 	unknownFilterSeparator tagFilterSeparator
 
 	// defaultFilterSeparator represents the default filter separator with no negation.
-	defaultFilterSeparator = tagFilterSeparator{str: ":", negate: false}
+	defaultFilterSeparator = tagFilterSeparator{
+		str: ":", negate: false,
+	}
+	databricksFilterSeparator = tagFilterSeparator{
+		str: "#", negate: false,
+	}
 
 	// validFilterSeparators represent a list of valid filter separators.
 	// NB: the separators here are tried in order during parsing.
 	validFilterSeparators = []tagFilterSeparator{
+		// The separators are sorted by priorities here. If one applies successfully, the ones below it are not considered.
+		databricksFilterSeparator,
```

Comment:
It is smart to add a new separator to replace `:`, one concern, promSQL use '#' as [comment](https://prometheus.io/docs/prometheus/latest/querying/basics/#comments), but I am thinking it shouldn't have conflict with our rule definition?

Reply:
will m3 filter based on `:` correctly? Can we do a unit test on that?

Reply:
> add a new separator to replace :

It's not to replace ':'. The logic here is to separate by '#' first. If it fails, try ':' then. Why do we try '#' first? It's to solve the following ambiguous case.
```
__name__#metric_name:10m
```
If we try ':' first, the case above will be parsed incorrectly.

Reply:
'#' is not an acceptable char for metric names and label names/values. Therefore, this change won't impact any existing aggregation rules.

<img width="864" alt="image" src="https://user-images.githubusercontent.com/2289363/226713113-0f5b41e2-e9ce-4e49-8e25-6cf304b9593a.png">

Reply:
sg, this is awesome!